KachnaOnline.App/Controllers/BoardGameController.cs

```diff
@@ -0,0 +1,354 @@
+// BoardGameController.cs
+// Author: František Nečas
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using KachnaOnline.Business.Exceptions;
+using KachnaOnline.Business.Exceptions.BoardGames;
+using KachnaOnline.Business.Facades;
+using KachnaOnline.Dto.BoardGames;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace KachnaOnline.App.Controllers
+{
+    [ApiController]
+    [Route("boardGames")]
+    public class BoardGameController : ControllerBase
+    {
+        private readonly BoardGameFacade _facade;
+
+        public BoardGameController(BoardGameFacade facade)
+        {
+            _facade = facade;
+        }
+
+        /// <summary>
+        /// Returns the current list of board games.
+        /// </summary>
+        /// <remarks>
+        /// The queries can be combined in any way.
+        /// </remarks>
+        /// <param name="categoryId">If present, only games from this category will be returned.</param>
+        /// <param name="players">If present, only games which can be played by this number of players will be
+        /// returned.</param>
+        /// <param name="available">If present, only games with this availability will be returned.</param>
+        /// <param name="visible">If present, only games with this visibility will be returned.
+        /// Regular and unauthenticated users will always receive only visible games.</param>
+        /// <returns>List of <see cref="BoardGameDto"/> corresponding to the given queries.</returns>
+        /// <response code="200">Returns the list of board games.</response>
+        [AllowAnonymous]
+        [ProducesResponseType(200)]
+        [Produces("application/json")]
+        [HttpGet]
+        public async Task<ActionResult<List<BoardGameDto>>> GetBoardGames(
+            [FromQuery] int? categoryId,
+            [FromQuery] int? players,
+            [FromQuery] bool? available,
+            [FromQuery] bool? visible)
```

Comment:
Isn't it better to be explicit for the sake of maintainability?

Reply:
No, this is simply redundant.

Reply:
Therefore, unless it is absolutely necessary to use `[FromQuery]`, `[FromBody]`, `[FromRoute]` etc. (to force the controller to bind to a specific parameter only in case of possible conflicts), we should refrain from using these bind attributes altogether?

Reply:
Yes. If it does what you expect it to do without explicitly specifying the binding source, then don't specify it. The default behaviour is typically what you want – if I simplify it a bit, basic types are bound from query or route, complex types are bound from the request body (by deserializing a JSON).